Backport from 0.6.x branch; apparently this hasn't even landed on
the stable 0.5.x branch yet.

diff -up wpa_supplicant-0.5.7/driver_wext.c.dynamic-wep-mac80211 wpa_supplicant-0.5.7/driver_wext.c
--- wpa_supplicant-0.5.7/driver_wext.c.dynamic-wep-mac80211	2007-10-28 00:57:11.000000000 -0400
+++ wpa_supplicant-0.5.7/driver_wext.c	2007-10-28 00:57:38.000000000 -0400
@@ -1724,6 +1724,7 @@ static int wpa_driver_wext_keymgmt2wext(
 {
 	switch (keymgmt) {
 	case KEY_MGMT_802_1X:
+	case KEY_MGMT_802_1X_NO_WPA:
 		return IW_AUTH_KEY_MGMT_802_1X;
 	case KEY_MGMT_PSK:
 		return IW_AUTH_KEY_MGMT_PSK;
